xarray/core/dataset.py

```diff
@@ -225,7 +231,9 @@ def _get_chunk(var, chunks):
         chunks.get(dim, None) or preferred_chunk_sizes
         for dim, preferred_chunk_sizes in zip(dims, preferred_chunk_shape)
     )
-    chunk_shape = da.core.normalize_chunks(
+
+    # TODO ideally replace this with non-dask version
+    chunk_shape = normalize_chunks(
```

Comment:
we could vendor it in `dask_array_compat`

Reply:
`dask_array_compat` hasn't existed for a little while, so I put it in a new file `daskvendor.py`. (My logic being that this isn't really for compatibility with dask arrays exactly.)